### PR TITLE
Fix search releases page

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://dimagi-dev.atlassian.net/browse/HI-498

Fixes 2 issues:
- Hitting <enter> in a search box just reloaded all pages with that widget without performing a search
- If you searched for a string on the releases page, elasticsearch would complain that it wasn't a number.

### CODE REVIEW

**Review by emord — APPROVED** [review]

File: corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_search_box.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <script type="text/html" id="ko-search-box-template">
     <div class="ko-search-box">
-        <form class="input-group">
+        <form class="input-group" data-bind="submit: clickAction">
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_search_box.html]: Why isn't this covered by the keypress handler on the input plus the click handler on the search button?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/app_manager/views/releases.py [hunk 2]
```diff
@@ -125,7 +125,10 @@ def _get_batch(start_build=None, skip=None):
             app_es = app_es.is_released()
         if query:
             app_es = app_es.add_query(build_comment(query), queries.SHOULD)
-            app_es = app_es.add_query(version(query), queries.SHOULD)
+            try:
+                app_es = app_es.add_query(version(int(query)), queries.SHOULD)
+            except ValueError:
```
Comment (orangejenny) [inline corehq/apps/app_manager/views/releases.py]: What was happening here? I swear I tested out text searches when implementing the version search...

Comment (proteusvacuum) [reply corehq/apps/app_manager/views/releases.py]: https://sentry.io/organizations/dimagi/issues/840150183

File: corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_search_box.html (hunk 1 above)
Comment (proteusvacuum) [reply corehq/apps/hqwebapp/templates/hqwebapp/partials/ko_search_box.html]: This is when the form is submitted (i.e. you hit <enter>). I think that's a different event. This was recently changed to be a `form` from a `div`, which I don't think has that action.

**Pull request merged**